src/javascript/binary/pages/trade/message.js

```diff
@@ -3,33 +3,97 @@
  * events based on type of response
  */
 var Message = (function () {
-    'use strict';
+    "use strict";
 
     var process = function (msg) {
-        var response = JSON.parse(msg.data);
-        if (response) {
-            var type = response.msg_type;
-            if (type === 'authorize') {
-                TUser.set(response.authorize);
-                TradeSocket.send({ payout_currencies: 1 });
-            } else if (type === 'active_symbols') {
+
+        var response;
+
+        try {
+            response  = JSON.parse(msg.data);
+            if("object" !== typeof response || !("msg_type" in response)) {
+                throw new Error("Response from WS API is not well formatted.");
+            }
+        } catch(e) {
+            throw new Error("Response from WS API is not well formatted.");
+        }
+
+        var msg_type = response.msg_type;
+        
+        switch(msg_type) {
+
+            case "error":
+
+                // change_password
+                if("change_password" in response.echo_req) {
+                    // e.g.: {"echo_req":{"old_password":"1111","new_password":"zzzzzzzz","change_password":"1"},"version":3,"error":{"code":"InputValidationFailed","message":"Input validation failed  - $.old_password: does not match the regex pattern ^.{5,25}$"},"msg_type":"error"}
+                    PasswordWS.apiResponse(response);
+                    break;
+                }
+
+                // code from old error case, please identify the called method
+                if(true) {
+                    $(".error-msg").text(response.error.message);
+                    break;
+                }
+
+                // still unknown?
+                console.log("An error occured.", response);
+                break;
+
+            case "authorize":
+
+                var thendo = "unknown";
+                if("passthrough" in response.echo_req && "thendo" in response.echo_req.passthrough) {
```

Comment:
On Portfolio page when the response to authorize arrives we want to update the balance figure. I thought other pages might need to perform their own processing upon authorization. To enable each caller perform its own desired actions, the caller can send an object in the passthrough like `{"thendo":"myThing"}`. So, thendo is understood to be like (1) first authorize, (2) then do myThing.